### PR TITLE
Image learner

### PR DESCRIPTION
**Pull request by qchiujunhao**
This PR adds a new Galaxy tool **ImageLearner** for image classification using the Ludwig framework. The tool supports training, evaluation, and report generation in a fully automated workflow, designed for ease of use by Galaxy users with image datasets.

## Features

- Pluggable architecture with a Ludwig backend (supporting resnet, efficientnet, vit, etc.) and we can add any backend (like autogluon) in the future easily!
- Input CSV support with required image_path, label, and optional split column
- Support for pretrained model weights and fine-tuning
- Automatic data splitting if no split column is provided
- YAML config generation by the script 
- Automatic metric extraction & visualization
- Rich HTML report with:
     - Config/parameters summary
     - Final training/validation/test metrics
     - Train/validation/test visualizations

## Inputs

- input_csv (required): CSV with columns image_path, label, and optionally split
- image_zip (required): ZIP archive of image files
- model_name: Choose from over 80 supported model backbones
- Optional tuning: epochs, batch size, learning rate, early stopping, etc.
- Optional random split proportions

## Outputs include:

- Trained model
- Report (HTML)
- Prediction CSV
- Experiment stats (JSON)
- Plots (PNG)

## Test Case
Includes a test using a subset of MNIST images (mnist_subset.csv + mnist_subset.zip) validating:
- HTML report content
- Predictions CSV formatting
- Proper file discovery for output collection

## Notes
- Handles fixed split (0/1/2 or 0/2) and generates validation split if needed
- YAML configuration is generated dynamically based on user input
- Backend is modular and can be replaced in the future if needed

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -175,7 +175,7 @@ jobs:
     - name: Install flake8
       run: pip install flake8 flake8-import-order
     - name: Flake8
```
[the remark this answers is not in the document]
Comment (qchiujunhao) [reply .github/workflows/pr.yaml:177]: better a bit this time
